File: client/form-pkh/src/app/formulardaten/form-e/einnahmen.ts
import {WeitereEinnahmenE} from './weitere-einnahmen-e';

export class Einnahmen {
  weitereEinnahmen: WeitereEinnahmenE[] = [];

  hatNichtselbstarbeit = false;
  nichtselbstarbeit = 0;

  hatSelbstarbeit = false;
  selbstarbeit = 0;

  hatVermietung = false;
  vermietung = 0;

  hatKapital = false;
  kapital = 0;

  hatWohngeld = false;
  wohngeld = 0;

  hatUnterhalt = false;
  unterhalt = 0;

  hatRente = false;
  rente = 0;

  hatArbeitslosg1 = false;
  arbeitslosg1 = 0;

  hatArbeitslosg2 = false;
  arbeitslosg2 = 0;

  hatKrankengeld = false;
  krankengeld = 0;

  hatElterngeld = false;
  elterngeld = 0;

  hatKindergeld = false;
  kindergeld = 0;
}

File: server/pkh-backend/src/main/java/pkh/form/materials/abschnitte/e/Einnahmen.java
package pkh.form.materials.abschnitte.e;

import java.util.List;

public class Einnahmen {
    private List<WeitereEinnahmen> weitereEinnahmen;

    private boolean hatNichtselbstarbeit;
    private String nichtselbstarbeit;

    private boolean hatSelbstarbeit;
    private String selbstarbeit;

    private boolean hatVermietung;
    private String vermietung;

    private boolean hatKapital;
    private String kapital;

    private boolean hatWohngeld;
    private String wohngeld;

    private boolean hatUnterhalt;
    private String unterhalt;

    private boolean hatRente;
    private String rente;

    private boolean hatArbeitslosg1;
    private String arbeitslosg1;

    private boolean hatArbeitslosg2;
    private String arbeitslosg2;

    private boolean hatKrankengeld;
    private String krankengeld;

    private boolean hatElterngeld;
    private String elterngeld;

    private boolean hatKindergeld;
    private String kindergeld;

    public Einnahmen(List<WeitereEinnahmen> weitereEinnahmen, boolean hatNichtselbstarbeit, String nichtselbstarbeit, boolean hatSelbstarbeit, String selbstarbeit,
                     boolean hatVermietung, String vermietung, boolean hatKapital, String kapital, boolean hatWohngeld, String wohngeld, boolean hatUnterhalt,
                     String unterhalt, boolean hatRente, String rente, boolean hatArbeitslosg1, String arbeitslosg1, boolean hatArbeitslosg2, String arbeitslosg2,
                     boolean hatKrankengeld, String krankengeld, boolean hatElterngeld, String elterngeld, boolean hatKindergeld, String kindergeld) {
        this.weitereEinnahmen = weitereEinnahmen;
        this.hatNichtselbstarbeit = hatNichtselbstarbeit;
        this.nichtselbstarbeit = nichtselbstarbeit;
        this.hatSelbstarbeit = hatSelbstarbeit;
        this.selbstarbeit = selbstarbeit;
        this.hatVermietung = hatVermietung;
        this.vermietung = vermietung;
        this.hatKapital = hatKapital;
        this.kapital = kapital;
        this.hatWohngeld = hatWohngeld;
        this.wohngeld = wohngeld;
        this.hatUnterhalt = hatUnterhalt;
        this.unterhalt = unterhalt;
        this.hatRente = hatRente;
        this.rente = rente;
        this.hatArbeitslosg1 = hatArbeitslosg1;
        this.arbeitslosg1 = arbeitslosg1;
        this.hatArbeitslosg2 = hatArbeitslosg2;
        this.arbeitslosg2 = arbeitslosg2;
        this.hatKrankengeld = hatKrankengeld;
        this.krankengeld = krankengeld;
        this.hatElterngeld = hatElterngeld;
        this.elterngeld = elterngeld;
        this.hatKindergeld = hatKindergeld;
        this.kindergeld = kindergeld;
    }
}
